include/MSI/AntSystem/Graph.h

```diff
@@ -10,25 +10,42 @@
 
 namespace msi::ant_system {
 
+struct Position {
+   Position(double _x1, double _y1, double _x2, double _y2);
+   double x1, y1, x2, y2;
+};
+
+struct Vertice {
+   double x, y, demand;
+};
+
 struct Edge {
+   Edge(double _pheromone, double _distance, Position _pos);
    double pheromone;
    double distance;
+   Position pos;
 
    [[nodiscard]] constexpr double prob() const noexcept {
       return std::pow(pheromone, g_alpha) * std::pow(1.0 / distance, g_beta);
    }
 };
 
 using VertId = std::size_t;
+using Table = std::vector<std::vector<double>>;
 
 class Graph {
+ public:
+   std::vector<Vertice> m_vertices;
+   std::vector<std::vector<double>> distance_table;
```

Comment:
m_distance_table

To z notacji węgierskiej. "m_" oznacza, że to jest class **m**ember. Od razu widać co to jest później w metodzie.

Reply:
ok, porawione